### ISSUE TITLE
Personal contact forms available even after disabling the "Use users' personal contact forms" permission

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#6064]
This was reported in a Zulip topic: https://backdrop.zulipchat.com/#narrow/stream/218635-Backdrop/topic/.E2.9C.94.20Personal.20contact.20form

> A site user just pointed out to me that the Personal contact form option is visible in their user account. I have disabled this under Permissions (Use users' personal contact forms) but I must be missing something. Can someone advise me please? Or is this a bug? I never use personal contact forms on my sites, so have never really looked at this before.

Might be related to #6063.

**Comment by kiamlaluno** [comment]
The *Contact* tab is always visible to users with the *administer users* permission. They could even the *Contact* tab somebody disabled for their own account, and they can also see their own *Contact* tab.

**Comment by klonos** [comment]
All that makes sense, however, I think that there should be a "Disable all personal contact forms" setting.

**Comment by kiamlaluno** [comment]
I guess it would make sense in the case a site plans to use the Contact module just for the site contact form.

What described in the issue summary could be working as expected. It is not clear which permissions that user account has, though, so I could not exclude cache is to blame.

**Comment by klonos** [comment]
> I guess it would make sense in the case a site plans to use the Contact module just for the site contact form.

Yup. And that feels like it should be a separate issue.

> What described in the issue summary could be working as expected. It is not clear which permissions that user account has, ...

Yes, I opened this issue to investigate if this is actually a bug (personal contact form being available even when the permission is not granted to a role).

**Comment by jenlampton** [comment]
> All that makes sense, however, I think that there should be a "Disable all personal contact forms" setting.

I have always wanted this for contact module. And it should be disabled by default! 

> Yes, I opened this issue to investigate if this is actually a bug

Have we confirmed that yet? I tested it and could not see the contact form as an editor with `View user profiles`. It does appear when I give them the `Administer user accounts` permission. @klonos can you ask if the person reporting that they could see the contact form had this permission?

**Comment by klonos** [comment]
> And it should be disabled by default!

Agreed 👍🏼 

> ...can you ask if the person reporting that they could see the contact form had this permission?

Will do 👍🏼

**Comment by olafgrabienski** [comment]
>> ... there should be a "Disable all personal contact forms" setting.

Let's try to avoid "Disable to enable" situations. If it turns out that we need the setting, it should be "Enable personal contact forms".

**Comment by klonos** [comment]
> ...there should be a "Disable all personal contact forms" setting.

Yes, that comment was before @jenlampton suggested to have the default OOTB be disabled. But also yes, "disable to enable" is bad UX, so good call 👍🏼

**Comment by klonos** [comment]
I've opened #6067 specifically for the new feature request. I'll try to reproduce the bug reported here next...

**Comment by kiamlaluno** [comment]
I would say it is expected that user administrators can contact people with an account on a site even if they disabled their contact form. Otherwise, how could user administrators contact any person with an account on a site about their account or how they use their account?

**Comment by kiamlaluno** [comment]
Accounts without the *administer users* permission cannot see the *Contact* tab. That is why the Drupal issue is specific about accounts with that permission.

**Comment by klonos** [comment]
> I would say it is expected that user administrators can contact people with an account on a site even if they disabled their contact form.

I have to disagree on this one @kiamlaluno. If a user has disabled that option, then it means that for whatever reason (mainly privacy I imagine) they do not like to be contacted via the site form - their preference should be respected. Administrators already have access to view the email used to register the user account, so they can still contact the owner of the account via a direct direct email if they really need to do that.

**Comment by kiamlaluno** [comment]
> If a user has disabled that option, then it means that for whatever reason (mainly privacy I imagine) they do not like to be contacted via the site form - their preference should be respected.

In that case, user administrators should not still contact that person, except in the case that person explicitly gave them the email address to use.

I am not also sure the *privacy* argument holds when there is the need to contact people who violated any term subscribed when creating the account on a site.

**Comment by klonos** [comment]
@kiamlaluno IF the person ever violates any term, then site admins/owners still have a way to contact them about it directly via their email. We cannot allow the fear of a potential _possibility_ to mandate allowing access to the person's contact form, despite their preference, just because something MAY happen. IF anything happens that warrants contacting the account owner, there is still a way to contact them - disabling the form according to the persons's preference does not stop that in any way. Whereas allowing access to that form by a group of people via permission permissions is against their preference (privacy issue, and also an additional potential security/spam vector if the site is compromised or permissions misconfigured), and the opposite of the setting the owner of the account has configured (a bug of the software).

**Comment by kiamlaluno** [comment]
> We cannot allow the fear of a potential _possibility_ to mandate allowing access to the person's contact form

Which issue does not allowing user administrators to use a user contact form resolve?  
It does not resolve any privacy nor security issue. On the contrary, using a user contact form, there is not track on my email application of the user email; the email address is not even stored for a short period of time on an external computer.

Speaking of privacy, should we also not allow users with the *bypass node access* to see unpublished nodes?

**Comment by izmeez** [comment]
I am confused as to what feedback is required for this issue now that #6067 has been merged. With the new feature if personal contact forms are not enabled an admin does not see the UI to send another user or themselves a personal contact, despite having the permission: `Use personal contact forms`.
